### DISCUSSION TITLE
Does SplitButton or DropDownButton have a preset behavior can be like ComboBox?

### DISCUSSION THREAD

**Discussion by tp1415926535** [fluentribbon/Fluent.Ribbon#1197] [Q&A] [answered]
It seems that _SplitButton_ or _DropDownButton_ doesn't have such a thing as selectedItem, if you want to display the selected item like ComboBox, you need to assign all the parameters of the selected item to the Button once, which is quite troublesome.

**Comment by batzen** [comment]
That's because they were not designed as selectors.
You could use a ComboBox if you'd want selection behavior.
What is your use case to use DropDown or SplitButton as selectors?

**Reply by tp1415926535** [reply]
For example, switching the pen of a drawing tool. There are pen, ballpoint, brush, etc. You can drop down to select an item as the current drawing way, but only the current pen name and icon are displayed

**Reply by batzen** [reply, marked as answer]
Ah, ok.
The best solution would be to re-style/re-template the ComboBox then.
